src/bare-metal/microcontrollers/examples/src/bin/mmio.rs

```diff
@@ -0,0 +1,64 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// ANCHOR: Example
+#![no_main]
+#![no_std]
+
+extern crate panic_halt as _;
+
+use core::mem::size_of;
+use cortex_m_rt::entry;
+
+/// GPIO port 0 peripheral address
+const GPIO_P0: usize = 0x5000_0000;
```

Comment:
I suppose there is a crate somewhere which would define this constant? But perhaps you're doing it by hand to show that there is no magic behind this?

Reply:
Oh, the next slide shows this. Nevermind :smile: